modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AbstractValidator.java

```diff
@@ -113,6 +116,26 @@ private static void validateSCIMObjectForRequiredSubAttributes(AbstractAttribute
                             }
                         }
                     }
+
+                    // Check for canonical attributes in groups.
+                    List<String> canonicalValues = subAttributeSchema.getCanonicalValues();
```

Comment:
can we extract this for a private method? WDYT?

Reply:
addressed, please check and confirm